openmc/universe.py

```diff
@@ -701,13 +705,24 @@ def get_unique_universes(self):
 
         """
 
-        unique_universes = np.unique(self._universes.ravel())
-        universes = {}
-
-        for universe in unique_universes:
-            universes[universe._id] = universe
-
-        return universes
+        univs = dict()
```

Comment:
Is `get_unique_universes` really that performance critical?  How many universes do you pack into your lattices?  It's hard to imagine that counting universes comes close to the performance requirements of handling all of the tallies on those universes.

If it really is a performance problem, maybe we should construct an array of unique universes in the `universes` setter method.